### ISSUE TITLE
'undefined' is not a function (evaluating 'this.onTransportActive.bind(this)')

### ISSUE THREAD

**Issue by mattheworiordan** [ably/ably-js#487]
Noticed this warning in our website error reporting system a few times over the last few months.  It happens rarely, but does seem to be a genuine JS script issue using version 1.x.

```
TypeError
'undefined' is not a function (evaluating 'this.onTransportActive.bind(this)')

Extract (which I don't quite trust):

40150>a.code};return d}(),H=function(){function e(a){if(!(this instanceof e))return new e(a);if(!a){var b="no options provided";c.logAction( {snip}
40012,400);if("*"===a.clientId)throw new q('Can\u2019t use "*" as a clientId as that string is reserved. (To change the default token reques {snip}
new h(this)}function h(a){this.rest=a;this.attached={}}var b=function(){};e.prototype.stats=function(a,c){void 0===c&&("function"==typeof a? {snip}
k.copy(k.defaultGetHeaders());this.options.headers&&k.mixin(e,this.options.headers);var f=this;C.get(this,function(a){return f.authority(a)+ {snip}
c||{};n=k.copy("get"==a?k.defaultGetHeaders():k.defaultPostHeaders(n));"string"!==typeof f&&(f=JSON.stringify(f));this.options.headers&&k.mi {snip}
this.attached[a]=c=new Y(this.rest,a,b);return c};h.prototype.release=function(a){delete this.attached[String(a)]};return e}(),J=function(){ {snip}
k.inherits(e,H);e.prototype.connect=function(){c.logAction(c.LOG_MINOR,"Realtime.connect()","");this.connection.connect()};e.prototype.close {snip}
"received event for non-existent channel: "+a)}};h.prototype.onTransportActive=function(){for(var b in this.all){var a=this.all[b];"attachin {snip}
closed:"detached",failed:"failed",suspended:"suspended"}[b],f;for(f in this.all){var l=this.all[f];k.arrIn(c,l.state)&&l.notifyState(e,a)}}; {snip}
h.prototype.onceNopending=function(b){if(this.hasNopending())b();else this.once("nopending",b)};h.prototype.hasNopending=function(){return k {snip}
b){y.call(this);this.ably=c;this.connectionManager=new P(c,b);this.state=this.connectionManager.state.state;this.recoveryKey=this.timeSerial {snip}
```

**Comment by funkyboy** [comment]
Looking at the logs this happens _only_ when Cloudflare (specifically with the User-Agent `Mozilla/5.0 (compatible; CloudFlare-AlwaysOnline/1.0; +http://www.cloudflare.com/always-online) AppleWebKit/534.34`) visits our home page. 
Requests come from the IP range `162.158.*` and happen roughly every 3-4 days.

**Comment by SimonWoolf** [comment]
> so if we decide to keep supporting IE8 we should fix that (relevant line is client/realtime.js line 32)

This wasn't actioned, so doing now: fixed by 7fbb039f3a62f8eae694eb15490e0dfafdc0f86a
